merge `sprint-3` in `main` branch

melzarjanchico-up opened the pull request:
**do not merge this yet**. i have to keep this open so that i can check if `sprint-3` is still compatible with `main` branch.

things that we have done this sprint (not yet duh)
- [x] display customer information
- [x] add customer information (@melzar-jan-chico-UP will focus on this hehe)
- [x] search customer

melzarjanchico-up committed:
moved user.dart to models

as it should sksk

melzarjanchico-up committed:
added horizontal padding in customer list cards

my ocd(not really) was triggered chz
+ i also set app screen to home screen for now

melzarjanchico-up committed:
moved all customer info related dart files

to a single folder (para mas sayun)
+ it should be working pls

melzarjanchico-up committed:
renamed user.dart to customer.dart

emma you are very krazy

melzarjanchico-up committed:
added `addCustomer` functionality (see comment)

note: pls ayaw ninyu iclick repeatedly ang customer icon sa appbar for customer list kay managhan si ariana grande. please. thank you.

DeinyRhed commented:
muchat ta ko na kaadbans ba

melzarjanchico-up committed:
arranged and renamed some files

i have to arrange and rename them kay kinda confusing na sya. i hope u understand huhu

melzar-jan-chico-UP commented:
> muchat ta ko na kaadbans ba

ayaw sag push dzae nag-arrange pako omg @DeinyRhed

melzarjanchico-up committed:
push for dayan customer list

makita ang customer list ani

melzarjanchico-up committed:
**(MAJOR)** changed `customer_list` code

made it so that front+back codes are separate...
take note...

vergeljabello committed:
added search button

search button added to customer and order list screen huhu

emmaacha committed:
added add_customer popup

emmaacha committed:
Merge branch 'sprint-3' of https://github.com/melzar-jan-chico-UP/dennis-lechon-crm into sprint-3

emmaacha committed:
addcustomer_popup w/ search customer

emmaacha committed:
fix testing error

melzarjanchico-up committed:
updated `customer_info.dart`

+ combined with emma's add customer popup
+ vor's scam na search

melzarjanchico-up committed:
seperated emma's `add_customer` pop-up

it worked and i dont know why... might be changed later

melzarjanchico-up committed:
updated `add customer` button

melzarjanchico-up committed:
added initial `add-customer`

note:
- dili pa ka-add ug tags
- birthday is still not addable

emmaacha committed:
Updated 'customer_screen' for add_customer popup

emmaacha committed:
fixed minor errors

DeinyRhed committed:
improved search functionalities

DeinyRhed committed:
for stash

DeinyRhed committed:
Merge branch 'sprint-3' of https://github.com/melzar-jan-chico-UP/dennis-lechon-crm into sprint-3

melzarjanchico-up committed:
customer list loading update

+ appbar is now included sa loading
+ i commented also a StreamBuilder() version for the loading since StreamProvider() still does the millesecond white screen thingy

emmaacha committed:
improved validators for add_customer

emmaacha committed:
fixed minor errors (added validators dependency)
